### ISSUE TITLE
Endless logout occurs if performed when device is connected to the network without access to internet (or poor connection)

### ISSUE THREAD

**Issue by pavloburykh** [status-im/status-react#18063]
**Steps:** 

1. Connect device to a network that has no access to internet (or simulate poor connection in emulator)
2. Login the app
3. Perform manual logout
4. See if user is redirected to login screen

**Actual result:** endless logout. User is not redirected to login screen. Sometimes I face endless login. But endless logout is easier to reproduce as it happens every time when following the above steps. 

[Status-debug-logs (87).zip](https://github.com/status-im/status-mobile/files/13545527/Status-debug-logs.87.zip)

https://github.com/status-im/status-mobile/assets/97245802/2d9b7d65-87bb-4194-9efa-137a53e6b07c

**Expected result:** user is redirected to login screen, logout is successful. 

### Additional Information

- Status version: nightly 
- Operating System: Android, iOS

**Comment by qfrank** [comment]
Hi @pavloburykh , can you reproduce this bug in airplane mode?

**Comment by qfrank** [comment]
I tried connect device to a network that has no access to internet with my iphone, but no luck to reproduce

**Comment by pavloburykh** [comment]
Hi @qfrank! Thank you for investigation.

> Hi @pavloburykh , can you reproduce this bug in airplane mode?

No, the bug is not reproducible in airplane mode.

> I tried connect device to a network that has no access to internet with my iphone, but no luck to reproduce

I have tried and noticed that indeed, initial bug with endless logout is not reproducible anymore. I have once managed to catch long logout on Android but not the endless one. 

At the same time I see another issue, this time related to endless login. 

### Unable to login the app after logout if user was connected to the network without access to internet

**Steps:** 

1. Connect device to a network that has no access to internet (or simulate poor connection in emulator)
2. Login the app
3. Perform manual logout
4. Perform login right after logout
5. See if user can successfully login

**Actual result:** user gets stuck on on login page and is not redirected into the account. User is able to login after app re-launching. 

[Status-debug-logs (94).zip](https://github.com/status-im/status-mobile/files/13872823/Status-debug-logs.94.zip)

https://github.com/status-im/status-mobile/assets/97245802/80edb356-6204-4761-bbe8-6853422d01df

**Expected result:** user is able to login. 

### Additional Information

- Status version: nightly 
- Operating System: Android, iOS

**Comment by qfrank** [comment]
> Hi @qfrank! Thank you for investigation.
> 
> > Hi @pavloburykh , can you reproduce this bug in airplane mode?
> 
> No, the bug is not reproducible in airplane mode.
> 
> > I tried connect device to a network that has no access to internet with my iphone, but no luck to reproduce
> 
> I have tried and noticed that indeed, initial bug with endless logout is not reproducible anymore. I have once managed to catch long logout on Android but not the endless one.
> 
> At the same time I see another issue, this time related to endless login.
> 
> ### Unable to login the app after logout if user was connected to the network without access to internet
> **Steps:**
> 
> 1. Connect device to a network that has no access to internet (or simulate poor connection in emulator)
> 2. Login the app
> 3. Perform manual logout
> 4. Perform login right after logout
> 5. See if user can successfully login
> 
> **Actual result:** user gets stuck on on login page and is not redirected into the account. User is able to login after app re-launching.
> 
> [Status-debug-logs (94).zip](https://github.com/status-im/status-mobile/files/13872823/Status-debug-logs.94.zip)
> 
>  telegram-cloud-document-2-5253892184477808420.mp4 
> **Expected result:** user is able to login.
> 
> ### Additional Information
> * Status version: nightly
> * Operating System: Android, iOS

Found something that could be related causing this issue although i didn't reproduce the endless, however, i can reproduce the explict delay when login->logout->login, i'll create a PR to see if it will work next Monday @pavloburykh

**Comment by pavloburykh** [comment]
> Found something that could be related causing this issue although i didn't reproduce the endless, however, i can reproduce the explict delay when login->logout->login, i'll create a PR to see if it will work next Monday @pavloburykh

Hey @qfrank! Many thanx for your investigation. Once the fix is ready - I will check it.
